src/main/java/be/syntra/devshop/DevshopBack/services/UserServiceImpl.java

```diff
@@ -32,4 +56,66 @@ public UserDto save(UserDto userDto) {
         userRepository.save(convertToUser(userDto));
         return userDto;
     }
+
+    /*
+     *@throws UserAlreadyRegisteredException
+     *@See DataIntegrityViolationException ->
+     * such Exception is thrown by three possible causes
+     * org.hibernate.exception.ConstraintViolationException
+     * org.hibernate.PropertyValueException
+     * org.hibernate.exception.DataException
+     * all of which have to do with persisting data to our database
+     * in our case, it'll happen when an email-address has already been registered
+     */
+    public User registerUser(String email, String password, String firstName, String lastName, List<UserRole> userRoles, Address address) throws UserAlreadyRegisteredException {
```

Comment:
use an object instead of all these parameters

Reply:
alt-enter mistake deriving from the controller. Fixed

Reply:
like registerDto has everything except the userRoles